lib/Interpreter/Value.cpp

```diff
@@ -231,9 +231,15 @@ namespace cling {
   namespace valuePrinterInternal {
     std::string printTypeInternal(const Value& V);
     std::string printValueInternal(const Value& V);
+    extern const char* const kUndefined;
   } // end namespace valuePrinterInternal
 
   void Value::print(llvm::raw_ostream& Out, bool Escape) const {
+    if (!m_Interpreter) {
+      Out << valuePrinterInternal::kUndefined << "\n";
```

Comment:
And the fact it needs to be declared in Value.cpp is not a unique problem i.e. `printTypeInternal` and `printValueInternal`.